### PR TITLE
[WIP] Modify get_node_names to return fully qualified names

### PR DESCRIPTION
**Pull request by jhdcs**
This pull request is joined to [https://github.com/ros2/rclcpp/pull/698](url). This is intended to modify the current tests to work with the new version of get_node_names.

### CODE REVIEW

File: test_rclcpp/test/node_check_names.cpp [hunk 1]
```diff
@@ -40,16 +43,23 @@ int main(int argc, char ** argv)
   int counter = 0;
   const std::chrono::steady_clock::time_point max_runtime =
     std::chrono::steady_clock::now() + std::chrono::seconds(10);
-  std::string node_to_look_for("node_with_name");
+
   while (rclcpp::ok()) {
-    printf("\n");
-    auto names = node->get_node_graph_interface()->get_node_names();
+    auto names_namespaces = node->get_node_graph_interface()->get_node_names_and_namespaces();
```
Comment (sloretz) [inline test_rclcpp/test/node_check_names.cpp]: Is `get_node_names_and_namespaces()` supposed to be `get_node_names()`?

Comment (jhdcs) [reply test_rclcpp/test/node_check_names.cpp]: I changed it to get_node_names_and_namespaces since the only thing that these functions are interested in is the name, not a fully qualified name. As such, I figured it's easier to grab the name from a tuple than extract from a string.

File: test_rclcpp/test/node_check_names.cpp [hunk 2]
```diff
@@ -40,16 +43,23 @@ int main(int argc, char ** argv)
...
+    std::vector<std::string> names;
+    std::transform(names_namespaces.begin(),
+      names_namespaces.end(),
+      std::back_inserter(names),
+      [](std::pair<std::string, std::string> nns) {return nns.first;}
+    );
+    printf("BEGIN DISCOVERED NODES\n");
```
Comment (wjwwood) [inline test_rclcpp/test/node_check_names.cpp]: Looks like debug messages, can you remove those please?

Comment (jhdcs) [reply test_rclcpp/test/node_check_names.cpp]: Whoops, on it.

**Comment by jhdcs** [comment]
> Taking a step back, what are these tests checking for? Maybe they should be updated to instead check for the full name? They may have only been checking for just the base name of the node because that's all the old behavior gave.

I think that checking for the full name is probably the best idea. But that will require me to make quite a few more changes. I'll get started.

**Comment by wjwwood** [comment]
Ok, sorry, wasn't trying to invent work for you, but I think it would be better to refresh these particular tests. Thanks.

**Comment by jhdcs** [comment]
No, no! It's A-OK! Though, I'm having a bit of difficulty updating the tests - it seems some of the test files are actually generated, and I'm not positive how to set up the generated node names. I'll push my latest changes in a sec.

**Comment by jhdcs** [comment]
I just bit the bullet and implemented the string modification in the test. We needed to cast argv[1] to a std::string anyways.

Is there a preferred way to do this? Can I get a CI on this?

**Comment by jhdcs** [comment]
@wjwwood Do the changes look acceptable?

File: .gitignore [hunk 3]
```diff
@@ -1 +1,5 @@
 *.pyc
+
+*.swp
+
+tags
```
Comment (sloretz) [inline .gitignore]: Mind removing the changes to `.gitignore`?

File: test_rclcpp/test/test_multithreaded.cpp [hunk 4]
```diff
@@ -268,6 +268,7 @@ static inline void multi_access_publisher(bool intra_process)
     .use_intra_process_comms(intra_process);
 
   auto node = rclcpp::Node::make_shared(node_topic_name, options);
+  printf("Node initialized");
```
Comment (sloretz) [inline test_rclcpp/test/test_multithreaded.cpp]: Mind removing this print statement? It seems unrelated.

File: test_rclcpp/test/test_multithreaded.cpp [hunk 5]
```diff
@@ -277,6 +278,7 @@ static inline void multi_access_publisher(bool intra_process)
 
   const size_t num_messages = 5 * std::min<size_t>(executor.get_number_of_threads(), 16);
   auto pub = node->create_publisher<test_rclcpp::msg::UInt32>(node_topic_name, num_messages);
+  printf("Publisher initialized");
```
Comment (sloretz) [inline test_rclcpp/test/test_multithreaded.cpp]: Mind removing this print statement? It seems unrelated.

File: test_rclcpp/test/node_check_names.cpp [hunk 6]
```diff
@@ -12,7 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <algorithm>
```
Comment (sloretz) [inline test_rclcpp/test/node_check_names.cpp]: It doesn't look like `<algorithm>` is being used in this file.

File: test_rclcpp/test/node_check_names.cpp [hunk 7]
```diff
@@ -12,7 +12,10 @@
...
 #include <string>
+#include <utility>
+#include <vector>
```
Comment (sloretz) [inline test_rclcpp/test/node_check_names.cpp]: Because `auto` is used and `vector` isn't I think this include could be omitted.

File: test_rclcpp/test/node_check_names.cpp [hunk 8]
```diff
@@ -12,7 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <algorithm>
 #include <string>
+#include <utility>
```
Comment (sloretz) [inline test_rclcpp/test/node_check_names.cpp]: I'm not sure if this header is being used explicitly in this file.

File: test_rclcpp/test/node_name_list.cpp [hunk 9]
```diff
@@ -12,7 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <algorithm>
 #include <string>
+#include <utility>
+#include <vector>
```
Comment (sloretz) [inline test_rclcpp/test/node_name_list.cpp]: Same comments about these headers

**Comment by sloretz** [comment]
CI (just `test_rclcpp` on windows since there are comments to address before full CI) [Build Status](https://ci.ros2.org/job/ci_windows/6683/)

File: .gitignore (hunk 3 above)
Comment (jhdcs) [reply .gitignore]: *Epic facepalm*

Sorry about that...

File: test_rclcpp/test/test_multithreaded.cpp (hunk 4 above)
Comment (jhdcs) [reply test_rclcpp/test/test_multithreaded.cpp]: Done

File: test_rclcpp/test/test_multithreaded.cpp (hunk 5 above)
Comment (jhdcs) [reply test_rclcpp/test/test_multithreaded.cpp]: Done

File: test_rclcpp/test/node_check_names.cpp (hunk 6 above)
Comment (jhdcs) [reply test_rclcpp/test/node_check_names.cpp]: Must have forgotten to remove it...

File: test_rclcpp/test/node_check_names.cpp (hunk 7 above)
Comment (jhdcs) [reply test_rclcpp/test/node_check_names.cpp]: I thought you still had to include any types that `auto` would need? Still, `vector` isn't being used anyways so out it goes.

File: test_rclcpp/test/node_check_names.cpp (hunk 8 above)
Comment (jhdcs) [reply test_rclcpp/test/node_check_names.cpp]: I'm not sure either. I'll cut it and see if it breaks.

File: test_rclcpp/test/node_name_list.cpp (hunk 9 above)
Comment (jhdcs) [reply test_rclcpp/test/node_name_list.cpp]: Got rid of 'em

**Comment by jhdcs** [comment]
Okay, made the changes you asked for. However, it looks like the testing API has gotten changed? Do I need to make any changes to account for this, or should it just work?

Because it seems like it's done something funky to my fastrps setup... The tests have suddenly stopped working.

**Comment by sloretz** [comment]
CI looks good https://github.com/ros2/rclcpp/pull/698#issuecomment-488407098
